seismic/inversion/wavefield_decomp/example_solvers.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python
+# coding: utf-8
+"""
+Examples of using the optimization (minimization) solver functions.
+"""
+
+import logging
+
+logging.basicConfig(format='%(asctime)s %(levelname)-8s %(message)s',
+                    level=logging.INFO,
+                    datefmt='%Y-%m-%d %H:%M:%S')
+
+import numpy as np
+from scipy.optimize import Bounds
+import matplotlib.pyplot as plt
+
+from seismic.inversion.wavefield_decomp.solvers import optimize_minimize_mhmcmc_cluster
+
+
+def bi_quadratic(x, mu, cov):
+    # The number returned from the function must be non-negative.
+    # The exponential of the negative of this value is the probablity.
+    x2fac = np.sqrt(np.matmul(np.matmul((x - mu).T, cov), x - mu))
```

Comment:
what happens if there is negative number?

Reply:
This number will always be non-negative, as it corresponds to an arbitrary quadratic with zero constant (i.e. minimum value is zero). Algebraically, I think this also follows from the Cauchy–Schwarz inequality.